### ISSUE TITLE
Unable to retrieve images, shows error URL signature mismatch

### ISSUE THREAD

**Issue by MeltedYeti** [theus/instantgram#48]
As the title says I can't seem to get the tool to work with Instagram any longer. Whenever I attempt to click on the bookmark to open the image I get a white page that just says, `URL signature mismatch`.

**Comment by theus** [comment]
Hi @MeltedYeti! Can you send an Instagram post link that you can't download?

**Comment by joe-wee** [comment]
This often happens when trying to download _video_ from stories - the domain it generates for the source file is typically in the rough format of:

https://instagram.LOCALE#-#.fna.fbcdn.net/v/ - where LOCALE seems to be an abbreviation of what datacenter to connect to, and the # are a combination of numbers like 1-3. For instance from San Diego my local code is FSAN, but from Los Angeles it switches to LAX.

If you get a URL mismatch error or the content won't load, you can replace the whole domain (everything before /v/) with:

https://scontent.cdninstagram.com/

And that seems to let you load / play / save the content.

Not sure if they have changed their URL format or what, but I've had 100% success rate on that _so far_.

I found it by fuskering the numbers in https://instagram.LOCALE#-#.fna.fbcdn.net - if you change them you usually get a DNS resolution error, but when you hit the back button in your browser it automatically reloads your previous URL as the "scontent.cdninstagram.com" domain.

**Comment by theus** [comment]
@joe-wee That's a good point! If this domain works as well with all media, I can replace it by default in [Instagram] and show only with this domain. What's your opinion?

**Comment by joe-wee** [comment]
I'm not able to do much rigorous testing right now as I'm in the middle of a move, but I edited my last comment because I realized some additional details.

The URL format that typically comes up when I use the bookmarklet is really:

`http(s):// instagram . **locale** /#-# . fna.fbcdn.net /v/ METADATA **&** METADATA **& LOCALE URL AGAIN &** METADATA`

the LOCALE code seems to switch all the time when I go through different exit nodes on my VPN - for San Diego it was FSAN but Los Angeles switches it to LAX.

For every piece of media I've been able to test, "scontent.cdninstagram.com/v/" seems to work in place of the initial locale in the base domain, but I have no idea if this works for other people in other places or not, and while that fixed my main issue with certain story content not loading, it does not really fix the URL Signature Mismatch error on its own. I RARELY get the URL signature mismatch error. More often than not, I don't get an error code, the story content just doesn't load at all, and that's the case where changing the base domain seems to make it work again.

The **metadata & metadata & metadata** part seems particularly relevant to the "URL signature mismatch" error - if you take a particular URL and remove various metadata chunks you get various error messages, typically either URL signature mismatch, or Bad URL timestamp.

In this example URL, everything after the first QUESTION MARK is metadata, and removing or changing the chunk within the asterisks generates a URL signature mismatch error:

`https://scontent.cdninstagram.com/v/t50.2886-16/98860387_140842551418862_2939961386848876533_n.mp4 **?** efg=eyJ2ZW5jb2RlX3RhZyI6InZ0c192b2RfdXJsZ2VuLjcyMC5zdG9yeS5kZWZhdWx0IiwicWVfZ3JvdXBzIjoiW1wiaWdfd2ViX2RlbGl2ZXJ5X3Z0c19vdGZcIl0ifQ **&_nc_ht=scontent-lax3-1.cdninstagram.com** &_nc_cat=102&_nc_ohc=lyZOdI8bFL8AX_XTvpE&edm=ANmP7GQBAAAA&vs=17847213644594802_3649332660&_nc_vs=HBkcFQAYJEdHTjk1QVh1SHo5MkdJQUFBUFgzZlFSTzE4d29idXFIQUFBQRUAAsgBACgAGAAbAYgHdXNlX29pbAExFQAAJtrvivaFgsY%2FFQIoAkMzLBdAFszMzMzMzRgSZGFzaF9iYXNlbGluZV8xX3YxEQB16AcA&oe=60ADDA9B&oh=5cccddccb38c0a5fc7ace25d281a9376&_nc_sid=276363`

In this case the metadata chunk is another locale-#-#.cdninstagram domain but changing THAT one to scontent.cdninstagram.com does not fix the error. There are a couple other metadata chunks that cause an error message when you remove or change them, but I don't know exactly which one would be involved in this user's case.

Sorry that's a huge wall of text and I don't know if any of it's particularly helpful, but let me know if there's something else I can do to help. I'm an IT guy, not a programmer, so my usefulness is probably limited here. haha
